common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.h

```diff
@@ -90,8 +92,8 @@ class mon_wlan_hal_dwpal : public base_wlan_hal_dwpal, public mon_wlan_hal {
 
     bool dwpal_set_scan_params_fg(const sScanCfgParams &params)
     {
-        if (dwpal_nl_cmd_set(m_radio_info.iface_name, LTQ_NL80211_VENDOR_SUBCMD_SET_SCAN_PARAMS,
-                             (unsigned char *)&params, sizeof(params))) {
+        if (!dwpal_nl_cmd_set(m_radio_info.iface_name, LTQ_NL80211_VENDOR_SUBCMD_SET_SCAN_PARAMS,
+                              reinterpret_cast<unsigned char *>(&params), sizeof(params))) {
```

Comment:
This is the reason this fails to build on RAX40 - I'm now very glad I changed it to GCC8.3.
This is also the reason we never want to use C style cast and instead use C++ style casts.
The issue here is that this cast removes constness of the params argument. This means that the scan params should not be const, or that `dwpal_nl_cmd_set` signature should be modified to accept const.

Reply:
Current signature for `dwpal_nl_cmd_set` is:
```C++
bool base_wlan_hal_dwpal::dwpal_nl_cmd_set(const std::string &ifname, unsigned int nl_cmd,
                                           unsigned char *vendor_data, size_t vendor_data_size)
```

Since this function proxy the so-called "vendor data" without modifying it, it should be changed to accept `const void*`. This will remove the need for casting at all.

```C++
bool base_wlan_hal_dwpal::dwpal_nl_cmd_set(const std::string &ifname, unsigned int nl_cmd,
                                           const void *vendor_data, size_t vendor_data_size)
```

Reply:
Although changing "vendor_data" signature to **(const void *)** seems like the good solution i still prefer to pass it as const unsigned char * and pass a non-const copy of it to the dwpal library.
The reason is that it still requires explicitly casting to **(const void *)** and back inside dwpal_nl_cmd_set (or leaving it as implicit cast for the compiler's decision) .
I believe casting to void * is less safer.